src/core/ext/transport/chttp2/transport/internal.h

```diff
@@ -489,7 +488,10 @@ struct grpc_chttp2_stream {
 
   grpc_chttp2_incoming_metadata_buffer metadata_buffer[2];
 
-  grpc_chttp2_incoming_frame_queue incoming_frames;
+  grpc_chttp2_incoming_byte_stream *incoming_frames;
+  gpr_mu buffer_mu; /* protects unprocessed_incoming_frames_buffer and
```

Comment:
I was thinking we want to have the logic of deframing on user thread instead of transport thread.